### DISCUSSION TITLE
inpainting model Conversion error

### DISCUSSION THREAD

**Discussion by dongzheng9** [Amblyopius/Stable-Diffusion-ONNX-FP16#12] [Q&A] [answered]
Thanks
i use 1.sd-v1-5-inpainting.ckpt  2.Anything-v3-inpainting.safetensors 3.GalaxyTimeMachine's Inpainting models They all have errors in the model conversion 
                                                                                For example
 (sd_env) E:\stable ddiffusion amd\Stable-Diffusion-ONNX-FP16>python conv_sd_to_onnx.py --model_path ".\b-inpainting.safetensors" --output_path "./model/b-inpainting" --ckpt-original-config-file v1-inference.yaml --fp16
global_step key not found in model
Traceback (most recent call last):
  File "E:\stable ddiffusion amd\Stable-Diffusion-ONNX-FP16\conv_sd_to_onnx.py", line 410, in <module>
    pl = load_pipeline_from_original_stable_diffusion_ckpt(
  File "E:\stable ddiffusion amd\Stable-Diffusion-ONNX-FP16\sd_env\lib\site-packages\diffusers\pipelines\stable_diffusion\convert_from_ckpt.py", line 949, in load_pipeline_from_original_stable_diffusion_ckpt
    unet.load_state_dict(converted_unet_checkpoint)
  File "E:\stable ddiffusion amd\Stable-Diffusion-ONNX-FP16\sd_env\lib\site-packages\torch\nn\modules\module.py", line 1671, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for UNet2DConditionModel:
        size mismatch for conv_in.weight: copying a param with shape torch.Size([320, 9, 3, 3]) from checkpoint, the shape in current model is torch.Size([320, 4, 3, 3]).

How do I solve the problem?  Currently in webui - inpainting I am using (onnx inpainting model)
git clone -b onnx https://huggingface.co/runwayml/stable-diffusion-inpainting   （It can be used normally）
How do I fix the error and convert more inpainting models  
Thanks

**Comment by Amblyopius** [comment, marked as answer]
Hi, a true inpainting model differs from a txt2img model. It has more channels (for the mask and the source image). If you start from .ckpt/.safetensors you'll need a different yaml. To simplify solving this, I have added a v1 inpaint yaml for models that don't come with one. Should allow you to import any v1 based inpaint model (which should be most of them).

**Reply by dongzheng9** [reply]
thx I made it👍

Look, man you're the most handsome guy 😊😊🌈
